### PR TITLE
Refactor st

### PR DESCRIPTION
**Pull request by shangabl**
### Motivation
SecureTunneling feature did not have unit tests. 

#### Change summary
Refactored SecureTunnelingFeature to improve testability and added unit tests.

### Testing
Added unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/tunneling/SecureTunnelingContext.cpp [hunk 1]
```diff
@@ -20,6 +20,8 @@ namespace Aws
             {
                 constexpr char SecureTunnelingContext::TAG[];
 
+                SecureTunnelingContext::SecureTunnelingContext() {}
```
Comment (RogerZhongAWS) [inline source/tunneling/SecureTunnelingContext.cpp:23]: Just asking, do we need this here if you define `SecureTunnelingContext()=default` in the header file? Right now SecureTunnelingContext.h and SecureTunnelingFeature.h do not explicitly define the constructor as default in the header, rather they do it in the source file.

File: source/tunneling/SecureTunnelingFeature.h [hunk 2]
```diff
@@ -114,6 +115,20 @@ namespace Aws
                      */
                     std::string GetEndpoint(const std::string &region);
 
+                    /**
+                     * \brief Get the IotSecureTunneling client
+                     */
+                    virtual std::shared_ptr<AbstractIotSecureTunnelingClient> getClient();
```
Comment (RogerZhongAWS) [inline source/tunneling/SecureTunnelingFeature.h:121]: I guess going with the same reasoning as the feedback posted in the jobs refactoring PR, we rename this to createClient

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -33,6 +33,11 @@ if (NOT EXCLUDE_JOBS)
     list(APPEND DC_SRC ${JOBS_SRC})
 endif ()
 
+if (NOT EXCLUDE_ST)
+    file(GLOB ST_SRC "../source/tunneling/*.cpp")
+    list(APPEND ST_SRC ${ST_SRC})
+endif ()
```
Comment (RogerZhongAWS) [inline test/CMakeLists.txt:39]: Why is this needed in addition to the statements at line 41?

File: source/tunneling/SecureTunnelingFeature.cpp [hunk 4]
```diff
@@ -186,19 +183,19 @@ namespace Aws
...
+                    string accessToken = response->ClientAccessToken->c_str();
 
-                    string region = response->Region->c_str();
-                    if (region.empty())
+                    if (!response->Region.has_value() || response->Region->empty())
                     {
                         LOG_ERROR(TAG, "region cannot be empty");
                         return;
                     }
+                    string region = response->Region.value().c_str();
```
Comment (RogerZhongAWS) [inline source/tunneling/SecureTunnelingFeature.cpp:198]: very minor: consistency between this and line 191, even though they accomplish the same thing.

**Review by RogerZhongAWS — COMMENTED** [review]
Again, nice job on the tests. We really needed these.

File: source/tunneling/SecureTunnelingContext.cpp (hunk 1 above)
Comment (shangabl) [reply source/tunneling/SecureTunnelingContext.cpp:23]: Oh yeah, that works. Fixed.

File: source/tunneling/SecureTunnelingFeature.h (hunk 2 above)
Comment (shangabl) [reply source/tunneling/SecureTunnelingFeature.h:121]: Good call... I actually thought I had done this.

File: test/CMakeLists.txt (hunk 3 above)
Comment (shangabl) [reply test/CMakeLists.txt:39]: Thanks, not sure how that got duplicated.
